diff --git a/src/stable-diffusion.cpp b/src/stable-diffusion.cpp
index dc0e3b7..664e2a2 100644
--- a/src/stable-diffusion.cpp
+++ b/src/stable-diffusion.cpp
@@ -175,10 +175,22 @@ public:
         backend = ggml_backend_cuda_init(0);
 #endif
 #ifdef SD_USE_METAL
+    if (!backend) {
         LOG_DEBUG("Using Metal backend");
         backend = ggml_backend_metal_init();
+    }
+#endif
+#ifdef SD_USE_OPENCL
+    if (!backend) {
+        LOG_DEBUG("Using OpenCL backend");
+        backend = ggml_backend_opencl_init();
+        if (!backend) {
+            LOG_WARN("Failed to initialize OpenCL backend");
+        }
+    }
 #endif
 #ifdef SD_USE_VULKAN
+    if (!backend) {
         LOG_DEBUG("Using Vulkan backend");
         size_t device          = 0;
         const int device_count = ggml_backend_vk_get_device_count();
@@ -206,18 +218,13 @@ public:
         if (!backend) {
             LOG_WARN("Failed to initialize Vulkan backend");
         }
-#endif
-#ifdef SD_USE_OPENCL
-        LOG_DEBUG("Using OpenCL backend");
-        // ggml_log_set(ggml_log_callback_default, nullptr); // Optional ggml logs
-        backend = ggml_backend_opencl_init();
-        if (!backend) {
-            LOG_WARN("Failed to initialize OpenCL backend");
-        }
+    }
 #endif
 #ifdef SD_USE_SYCL
+    if (!backend) {
         LOG_DEBUG("Using SYCL backend");
         backend = ggml_backend_sycl_init(0);
+    }
 #endif
 
         if (!backend) {
