diff --git a/src/3rd_party/CLI/App.hpp b/src/3rd_party/CLI/App.hpp
--- a/src/3rd_party/CLI/App.hpp
+++ b/src/3rd_party/CLI/App.hpp
@@ -1688,14 +1688,14 @@ namespace detail {
 struct AppFriend {
 
     /// Wrap _parse_short, perfectly forward arguments and return
     template <typename... Args>
     static auto parse_arg(App *app, Args &&... args) ->
-        typename std::result_of<decltype (&App::_parse_arg)(App, Args...)>::type {
+        std::invoke_result_t<decltype (&App::_parse_arg), App, Args...> {
         return app->_parse_arg(std::forward<Args>(args)...);
     }
 
     /// Wrap _parse_subcommand, perfectly forward arguments and return
     template <typename... Args>
     static auto parse_subcommand(App *app, Args &&... args) ->
-        typename std::result_of<decltype (&App::_parse_subcommand)(App, Args...)>::type {
+        std::invoke_result_t<decltype (&App::_parse_subcommand), App, Args...> {
         return app->_parse_subcommand(std::forward<Args>(args)...);
     }
 };
